hivelocity/instancesV2_test.go

```diff
@@ -110,90 +100,125 @@ func standardMocks(m *mocks.API) {
 			OrderId:                  0,
 			OsName:                   "",
 			ProductId:                0,
+			Tags:                     []string{"instance-type=bare-metal-x"},
 		},
 		nil)
 
-	m.On("GetBareMetalDeviceIdResource", int32(9999999)).Return(
+	m.On("GetBareMetalDevice", mock.Anything, int32(9999999)).Return(
 		nil, client.ErrNoSuchDevice)
 }
 func Test_InstanceExists(t *testing.T) {
-	i2, m := newHVInstanceV2(t)
-	node := newNode()
-	ctx := context.Background()
-	standardMocks(m)
-	myBool, err := i2.InstanceExists(ctx, node)
-	require.NoError(t, err)
-	require.Equal(t, true, myBool)
-
-	node.Spec.ProviderID = "9999999"
-	myBool, err = i2.InstanceExists(ctx, node)
-	require.Equal(t, false, myBool)
-	require.NoError(t, err)
+	m := mocks.NewAPI(t)
+	i2, node, ctx := newTestData(m)
```

Comment:
I would define the context outside this function, as it is not really related to "test data". However, if you think that this is better, keep it.
I think I would actually give all functions that need it and where it doesn't matter in the test directly a context.Background().